### ISSUE TITLE
Could not resolve dev.icerock.moko:mvvm:0.6.0

### ISSUE THREAD

**Issue by GabrielNoam** [icerockdev/moko-mvvm#41]
I am combining it into my project after follow basic guidelines for building multi-project using:
https://proandroiddev.com/kotlin-multiplatform-very-beginners-guide-part-1-6419f74afa0f
https://proandroiddev.com/kotlin-multiplatform-very-beginners-guide-part-2-api-d54f7326dc57
I run into strange error of "Unable to find a matching variant of dev.icerock.moko:mvvm:0.6.0" (see below gradle output) thaw I am doing the exact steps in https://github.com/icerockdev/moko-mvvm README

I've added my project gradle files (app and shared)  to share location on my Google drive: 
https://drive.google.com/file/d/1Iq8IxXFnnnZE-Kpgb9xgbIMcmMvfLDqG/view?usp=sharing 

Thank you in advance for the great work.

Best regards,
Gabriel 

* What went wrong:
Execution failed for task ':Shared:compileKotlinAndroid'.
> Could not resolve all files for configuration ':Shared:androidCompileClasspath'.
   > Could not resolve dev.icerock.moko:mvvm:0.6.0.
     Required by:
         project :Shared
      > Unable to find a matching variant of dev.icerock.moko:mvvm:0.6.0:
          - Variant 'android-debugApiElements' capability dev.icerock.moko:mvvm:0.6.0:
              - Incompatible attribute:
                  - Required org.jetbrains.kotlin.platform.type 'jvm' and found incompatible value 'androidJvm'.
              - Other attributes:
                  - Found com.android.build.api.attributes.BuildTypeAttr 'debug' but wasn't required.
                  - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
                  - Found com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' but wasn't required.
                  - Found org.gradle.status 'release' but wasn't required.
                  - Required org.gradle.usage 'java-api' and found compatible value 'java-api'.
          - Variant 'android-debugRuntimeElements' capability dev.icerock.moko:mvvm:0.6.0:
              - Incompatible attribute:
                  - Required org.jetbrains.kotlin.platform.type 'jvm' and found incompatible value 'androidJvm'.
              - Other attributes:
                  - Found com.android.build.api.attributes.BuildTypeAttr 'debug' but wasn't required.
                  - Found com.android.build.api.attributes.VariantAttr 'debug' but wasn't required.
                  - Found com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' but wasn't required.
                  - Found org.gradle.status 'release' but wasn't required.
                  - Required org.gradle.usage 'java-api' and found compatible value 'java-runtime'.
          - Variant 'android-releaseApiElements' capability dev.icerock.moko:mvvm:0.6.0:
              - Incompatible attribute:
                  - Required org.jetbrains.kotlin.platform.type 'jvm' and found incompatible value 'androidJvm'.
              - Other attributes:
                  - Found com.android.build.api.attributes.BuildTypeAttr 'release' but wasn't required.
                  - Found com.android.build.api.attributes.VariantAttr 'release' but wasn't required.
                  - Found com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' but wasn't required.
                  - Found org.gradle.status 'release' but wasn't required.
                  - Required org.gradle.usage 'java-api' and found compatible value 'java-api'.
          - Variant 'android-releaseRuntimeElements' capability dev.icerock.moko:mvvm:0.6.0:
              - Incompatible attribute:
                  - Required org.jetbrains.kotlin.platform.type 'jvm' and found incompatible value 'androidJvm'.
              - Other attributes:
                  - Found com.android.build.api.attributes.BuildTypeAttr 'release' but wasn't required.
                  - Found com.android.build.api.attributes.VariantAttr 'release' but wasn't required.
                  - Found com.android.build.gradle.internal.dependency.AndroidTypeAttr 'Aar' but wasn't required.
                  - Found org.gradle.status 'release' but wasn't required.
                  - Required org.gradle.usage 'java-api' and found compatible value 'java-runtime'.
          - Variant 'iosArm64-api' capability dev.icerock.moko:mvvm:0.6.0:
              - Incompatible attributes:
                  - Required org.gradle.usage 'java-api' and found incompatible value 'kotlin-api'.
                  - Required org.jetbrains.kotlin.platform.type 'jvm' and found incompatible value 'native'.
              - Other attributes:
                  - Found org.gradle.status 'release' but wasn't required.
                  - Found org.jetbrains.kotlin.native.target 'ios_arm64' but wasn't required.
          - Variant 'iosX64-api' capability dev.icerock.moko:mvvm:0.6.0:
              - Incompatible attributes:
                  - Required org.gradle.usage 'java-api' and found incompatible value 'kotlin-api'.
                  - Required org.jetbrains.kotlin.platform.type 'jvm' and found incompatible value 'native'.
              - Other attributes:
                  - Found org.gradle.status 'release' but wasn't required.
                  - Found org.jetbrains.kotlin.native.target 'ios_x64' but wasn't required.
          - Variant 'metadata-api' capability dev.icerock.moko:mvvm:0.6.0:
              - Incompatible attributes:
                  - Required org.gradle.usage 'java-api' and found incompatible value 'kotlin-api'.
                  - Required org.jetbrains.kotlin.platform.type 'jvm' and found incompatible value 'common'.
              - Other attribute:
                  - Found org.gradle.status 'release' but wasn't required.

**Comment by Alex009** [comment]
hi!
can you check please versions of environment:
- kotlin 1.3.70 
- gradle 5.6.4+

also show please how you configure dependency?

**Comment by GabrielNoam** [comment]
Hello. Sorry for the late reply.
First, I've continued to work on migrating project from  moko-mvvm-0.6.0
to  moko-mvvm-0.3.1 and finally succeeded :)
I am using: distributionUrl=https\://
services.gradle.org/distributions/gradle-5.6.4-all.zip

   - minSdkVersion 24
   - targetSdkVersion 29
   - kotlin_version = '1.3.70'
   - ext.moko_mvvm_version = '0.6.0'

In app gradle
dependencies: implementation("dev.icerock.moko:mvvm:$moko_mvvm_version")
In shared module dependencies:
sourceSets {
        commonMain.dependencies {
            api 'org.jetbrains.kotlin:kotlin-stdlib-common'
            // MOKO - MVVM
            implementation "dev.icerock.moko:mvvm:$moko_mvvm_version"
            ....
         }
         ....
}

The problem with old project that did not compile was that there was a
mixture of versions
https://drive.google.com/file/d/1Iq8IxXFnnnZE-Kpgb9xgbIMcmMvfLDqG/view?usp=sharing

I think issue can be closed.
Thank you for all your efforts and dedication to assist.

**Comment by alan-camilo** [comment]
What was the issue exactly and what changes did you make to solve the issue? I have the exact same one trying to use moko mvvm on my multiplatform project created from scratch using kotlinlang tutorial.

root build.gradle : https://github.com/alan-camilo/paris-respire-app/blob/icerockmvvm/android/build.gradle.kts
mpp-module build.gradle.kts : https://github.com/alan-camilo/paris-respire-app/blob/icerockmvvm/module-mpp/build.gradle.kts

TLDR;
kotlin version 1.3.70
gradle version 6.2.2
settings.gradle has enableFeaturePreview("GRADLE_METADATA")

**Comment by GabrielNoam** [comment]
Dear Alan,
I've cloned your project in order to fix it and was unable to
build: paris-respire-app output ... not found.
Anycase I've updated Javier Arroyo jarroyoesp@gmail.com code example with
moko 0.6.0 but couldn't make a pull request :(
Therefore, please download the project from this link for now:
https://drive.google.com/open?id=1_T0atDRW-jLbkMH1QAaLvcW7xN3yxjDp

Make sure you are using *Android Studio 3.6 RC 3*

Hope this helps.
Best,
Gabriel

**Comment by alan-camilo** [comment]
@GabrielNoam thx for uploading your project, it helped me to understand my mistake.

My multiplatform module (module-mpp) was a jvm lib, I migrate it to an android lib. Now the build.gradle.kts of the module-mpp looks like this: https://github.com/alan-camilo/paris-respire-app/blob/dev/module-mpp/build.gradle.kts

You can note that in the kotlin section, the target set-up went from 
```
jvm("android")
```
to 
```
targets {
        targetFromPreset(presets.getByName("android"), "android")
    }
```

It now compiles without error.

**Comment by GabrielNoam** [comment]
Greate :)
Thanks
